### PR TITLE
Subscription headers refactor

### PR DESCRIPTION
**Pull request by dankraw**
Refactoring headers from `Map<String, String>` to `List<Header>`.

### CODE REVIEW

**Comment by wojtkiewicz** [comment]
👍

**Comment by pawelszymczyk** [comment]
👍

**Comment by chemicL** [comment]
👍

File: hermes-api/src/main/java/pl/allegro/tech/hermes/api/Header.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getValue() {
+        return value;
+    }
+}
```
Comment (jmnarloch) [inline hermes-api/src/main/java/pl/allegro/tech/hermes/api/Header.java]: What about hashCode & equals?

Comment (dankraw) [inline hermes-api/src/main/java/pl/allegro/tech/hermes/api/Header.java]: added, thx

**Comment by dankraw** [comment]
moved to release-0.8.8 branch

**Pull request closed without merging**